src/main/java/ca/serum390/godzilla/api/handlers/InventoryHandler.java

```diff
@@ -45,4 +54,94 @@ private InventoryHandler() {}
                              .with(DESCRIPTION, "A finished bicycle.")
                     ));
     }
+    */
+
+    /**
+     * {@link InventoryRepository}
+     */
+    private final InventoryRepository items;
+
+    public InventoryHandler(InventoryRepository items){
+        this.items = items;
+    }
+
+    /**
+     * Insert into the table
+     */
+    public Mono<ServerResponse> insert(ServerRequest req){
+        return req.bodyToMono(Item.class)
+            .flatMap(items::save)
+            .flatMap(id -> created(URI.create("/items/" + id)).build());
+    } 
+
+    /**
+     * Update item in the table
+     */
+
+    /**
+     * Get db table inventory
+     */
+    // public Mono<ServerResponse> getAll(ServerRequest req){
+    //     return ok().body(
+    //         items.findAll(),
+    //         Item.class);
+    // }
+
+    /**
+     * Get db table inventory by id 
+     */
+    public Mono<ServerResponse> getById(ServerRequest req){
+        return items.findById(
+            Integer.parseInt(req.pathVariable("id")))
+            .flatMap(inventory -> ok().body(Mono.just(inventory),Item.class))
+            .switchIfEmpty(notFound().build()
+        );
+    }
+
+    /**
+     * Delete element in table by id
+     */
+    public Mono<ServerResponse> deleteByID(ServerRequest req){
+        return items.deleteById(
+            Integer.parseInt(req.pathVariable("id")))
+            .flatMap(deleted -> noContent().build()
+        );
+    }
+
+    /**
+     * Find by item name
+     */
+    public Mono<ServerResponse> findbyName(ServerRequest req) {
+        String name = req.pathVariable("item_name");
+        return ok().contentType(APPLICATION_JSON).body(items.findbyName(name),Item.class);
+    }
+
+    /**
+     * 
+     */
+    public Mono<ServerResponse> getBy(ServerRequest req) {
+        Optional<String> name = req.queryParam("name");
+        Optional<String> id = req.queryParam("id");
+
+        if (name.isPresent()) {
+            // Query by name
+            return ok().contentType(APPLICATION_JSON).body(items.findbyName(name.get()), Item.class);
```

Comment:
call findByName(ServerRequest) to avoid duplicate code. same thing for the id

Reply:
I would delete those other methods or change them to accept a String instead of a ServerRequest and pass them the parsed query string param in the `getBy` method. Right now those other methods are trying to parse pathVariable, but the new method `getBy` is checking query string, so we wouldn't be able to just send them the ServerRequest

Reply:
changed